### PR TITLE
Various task definition improvements

### PR DESCRIPTION
**Pull request by colinking**
## Description

Unwraps the `timeout` and `resources` fields into `int` and `map` values, respectively. This makes it easier to construct a `Definition` struct. The structs are no longer necessary (e.g. there used to be some default timeout behavior but that was removed).

Marks a few fields in the JSON schema as non-required. These haven't been required by our API, but for some reason were marked as required in the schema.

Tweaks the serialization of options to not use label objects when not needed. This is useful in the JS transformer (see upstream).

For more context, see comments in the PR these changes were extracted from: https://github.com/airplanedev/lib/pull/590

## Test plan

I tested:
1. I can deploy a task with resources, options, and timeouts
2. I can initialize an inline and YAML-based JS task
3. The studio can discover/dev inline JS tasks

### CODE REVIEW

**Comment by colinking** [comment]
Current dependencies on/for this PR:
* main
  * **PR #606** <a href="https://app.graphite.dev/github/pr/airplanedev/lib/606" target="_blank">Graphite</a>  👈
    * **PR #590** <a href="https://app.graphite.dev/github/pr/airplanedev/lib/590" target="_blank">Graphite</a> 
      * **PR #591** <a href="https://app.graphite.dev/github/pr/airplanedev/lib/591" target="_blank">Graphite</a> 
        * **PR #592** <a href="https://app.graphite.dev/github/pr/airplanedev/lib/592" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/airplanedev/lib/606?utm_source=stack-comment).

File: pkg/runtime/javascript/fixtures/transformer/parameters.out.airplane.js [hunk 1]
```diff
@@ -11,7 +11,7 @@ export default airplane.task(
         required: false,
         default: "My default",
         regex: "^.*$",
-        options: ["Thing 1", "Thing 2", {
```
Comment (ameyakhare) [inline pkg/runtime/javascript/fixtures/transformer/parameters.out.airplane.js:14]: why is this removed? we should be able to have options without labels right?

**Review by ameyakhare — APPROVED** [review]

File: pkg/runtime/javascript/fixtures/transformer/parameters.out.airplane.js (hunk 1 above)
Comment (colinking) [reply pkg/runtime/javascript/fixtures/transformer/parameters.out.airplane.js:14]: Because of the change to option marshaling, it broke that specific JS transformer test. I didn't bother to backport the fix from the upstream PR since it would have created a bunch of merge conflicts and the JS transformer isn't used by customers yet.
